### PR TITLE
New config fix

### LINKED ISSUE #998: Problem with configuration system
A bug has been introduced in the configuration system which causes out-of-date values to be locked into a users `.gangarc` file. Two instances of this are
   * Starting Ganga as ganga -o'[Logging]Ganga.Lib=DEBUG' will if it at the same time is a release that the user has not used before
   * Values that are changed in `.INI` file are likewise locked into the users .gangarc when the `,gangarc` file is recreated with a new release.

As an example, my `.gangarc` file at Imperial has the values below uncommented

```
load_default_Dirac_backend = False
splitFilesChunks = 5000
datasets = string_dataset_shortcut
associate = {'log': 'TextMerger', 'sim': 'LHCbFileMerger', 'text': 'TextMerger', 'dst': 'LHCbFileMerger', 'txt': 'TextMerger', 'root': 'RootMerger', 'mdst': 'LHCbFileMerger', 'digi': 'LHCbFileMerger'}
creds_poll_rate = 86400
arch = x86_64-slc6-gcc49-opt
location = /cvmfs/lhcb.cern.ch/lib/RootConfig
path = /cvmfs/lhcb.cern.ch/lib/RootConfig/pro/x86_64-slc6-gcc49-opt
pythonhome = /cvmfs/lhcb.cern.ch/lib/GangaConfig/python/pro/x86_64-slc6-gcc49-opt
pythonversion = pro
version = pro
os.environ["PATH"]+=":."
preexecute = exec('import os\nos.chdir(os.environ["TMPDIR"])\nos.environ["PATH"]+=":."\nos.environ["HOME"]=os.environ["TMPDIR"]\nimport shutil,pwd\nshutil.copy(os.path.join("/vols/lhcbdisk05",pwd.getpwuid(os.getuid())[0],".proxy","proxy"),"/tmp/x509up_u%d" % os.getuid())')
submit_str = /home/hep/lhcb/bin/batchproxy; cd %s; qsub -S /usr/bin/python -cwd -V %s %s %s %s
encodeDefaultProxyFileName = False
group = lhcb_user
options = []
submit_with_threads = False
pre_script = ['source /afs/cern.ch/lhcb/software/releases/LBSCRIPTS/prod/InstallArea/scripts/LbLogin.sh', 'source `runpy LbConfiguration.SetupProject --shell=sh --mktemp Ganga`']
queue = hepshort.q
```

The code should be made such that the only uncommented lines written into a new `.gangarc` are those that were uncommented in the file already. A method for cleaning up the mess created in users' `.gangarc` files should also be developed.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #998 

I added the gangarc set of option values and updated the code that updates/generates a new gangarc to only use those.

### CODE REVIEW

**Comment by mesmith75** [comment]
Can someone take a look and then we can merge this?

File: python/Ganga/GPIDev/Lib/Config/Config.py [hunk 1]
```diff
@@ -266,10 +266,14 @@ def config_file_as_text():
                                 def_value = "\n# ".join(def_lines)
                         except AttributeError as err:
                             pass
+                    print 'o: ', o, ' def_value: ', def_value, ' value: ', value
```
Comment (alexanderrichards) [inline python/Ganga/GPIDev/Lib/Config/Config.py]: Did you mean to leave a print statement in here?

Comment (egede) [inline python/Ganga/GPIDev/Lib/Config/Config.py]: Did realise this was a print (and not a logging) statement. It was quite helpful when running that this was printed out (it is actually half the way towards the system that we discussed for moving values forward). Should probably just be deleted for now.

Comment (mesmith75) [inline python/Ganga/GPIDev/Lib/Config/Config.py]: Yes I'll get rid of that for now and put it in properly for copying over the gangarc.

**Pull request merged**